src/httpfetch.cpp

```diff
@@ -766,6 +767,27 @@ void httpfetch_sync(const HTTPFetchRequest &fetch_request,
 	fetch_result = *ongoing.complete(res);
 }
 
+bool httpfetch_sync_interruptible(const HTTPFetchRequest &fetch_request,
+		HTTPFetchResult &fetch_result, long interval)
+{
+	if (Thread *thread = Thread::getCurrentThread()) {
```

Comment:
thought: does clientlauncher/guiEngine even run in a `Thread`? I remember seeing code that used `porting::signal_handler_killstatus` and other custom ways to abort.

(we might have to create a `Thread` object for the main thread...)

Reply:
I think not, but all the HTTP requests happen in `core.handle_async` calls, which are `Thread`s, so it doesn't matter in practice (except for the "get CDB package dependencies" request)

Reply:
oh true, for some reason I forgot about the async thing.